tracer/transport.go

```diff
@@ -153,7 +183,10 @@ func (t *httpTransport) SendServices(services map[string]Service) (*http.Respons
 		return t.SendServices(services)
 	}
 
-	response.Body.Close()
+	if sc := response.StatusCode; sc != 200 {
+		return response, fmt.Errorf("SendServices expected response code 200, received %v", sc)
```

Comment:
Following the code path, we hit the [`logError`](https://github.com/DataDog/dd-trace-go/blob/7bdebb245d4cda39ed54ff7b97fcd4ece295bf11/tracer/errors.go#L143-L157) function that limits the number of logs that are written. Also this isn't the hot-path of our agent, so the number of occurrences are limited.

I think we're fine here, considering that we're expecting a `200` but users are notified for any wrong state.